machine/quanta/quanta_common_p2020/rootconf/sysroot-init/machine-detect.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+platform=$(/bin/sed 's/quanta,//g' /proc/device-tree/compatible)
+cp -f /etc/machine.conf /etc/machine_common_p2020.conf
+/bin/sed "s/common_p2020/${platform}/g" /etc/machine_common_p2020.conf > /etc/machine.conf
+
```

Comment:
Are there any ways this can fail?  What happens if $platform for some reason is blank?

You have added more moving parts here.  So I'm just trying to think about what could go wrong.